includes/manifest.inc

```diff
@@ -55,18 +57,36 @@ function islandora_iiif_manifests_create_manifest_json(AbstractObject $object, $
 
 /**
  * Response in JSON for the canvas.
+ * @return void
  */
 function islandora_iiif_manifests_create_canvas_json(AbstractObject $object) {
   
   module_load_include('inc', 'islandora', 'includes/utilities');
   
   $output = islandora_iiif_manifests_build_canvas($object);
-  
+  $output['@context'] = 'http://iiif.io/api/presentation/2/context.json';
   islandora_iiif_manifests_create_json_response('200 OK', $output);
 }
 
 /**
- * Build manifest array.
+ * Send a JSON-encoded response for the image-painting annotation.
+ * There should be exactly one such annotation in our Canvas, allowing us
+ * to build the Canvas and taking the image annotation from it.
+ * @return void
+ */
+function islandora_iiif_manifests_create_annotation_json(AbstractObject $object) {
+  
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  
+  $output = islandora_iiif_manifests_build_canvas($object);
+  $annotation = $output['images'][0];
```

Comment:
Shouldn't you check existence first?

Reply:
` // Check if an image is available
  if (isset($output['images'][0])) {
    $annotation = $output['images'][0];
    $annotation['@context'] = 'http://iiif.io/api/presentation/2/context.json';
  }
  else {
    $annotation = array(
      "data" => '{"error": "No manifest available"}',
      "status" => "404 Not Found"
    );
  }
  
  islandora_iiif_manifests_create_json_response('200 OK', $annotation);`